components/server/src/signal_handler.rs

```diff
@@ -4,15 +4,14 @@ pub use self::imp::wait_for_signal;
 
 #[cfg(unix)]
 mod imp {
-    use engine_rocks::RocksEngine;
-    use engine_traits::{Engines, MiscExt, RaftEngine};
+    use engine_traits::{Engines, KvEngine, MiscExt, RaftEngine};
     use libc::c_int;
     use nix::sys::signal::{SIGHUP, SIGINT, SIGTERM, SIGUSR1, SIGUSR2};
     use signal::trap::Trap;
     use tikv_util::metrics;
 
     #[allow(dead_code)]
-    pub fn wait_for_signal<ER: RaftEngine>(engines: Option<Engines<RocksEngine, ER>>) {
+    pub fn wait_for_signal<ER: RaftEngine>(engines: Option<Engines<impl KvEngine, ER>>) {
```

Comment:
if the EK parameter is made `impl KvEngine` i think ER can be reduced to `impl RaftEngine` too? 🤔